### PR TITLE
Add possibility to disable gogs reverseproxy authentication/registration

### PR DESCRIPTION
**Pull request by dabelenda**
To avoid changing the behavior when not defined, the variables are inverted.

### CODE REVIEW

File: templates/app.ini.tmpl [hunk 1]
```diff
@@ -18,8 +18,19 @@ REVERSE_PROXY_AUTHENTICATION_USER = X-Forwarded-User
 
 [service]
 ENABLE_NOTIFY_MAIL                      = {{getenv "GOGS_SERVICE_ENABLE_NOTIFY_MAIL"}}
+{{if not ( eq ( getenv "GOGS_SERVICE_DISABLE_REVERSE_PROXY_AUTHENTICATION" ) "" ) }}
```
Comment (mcanevet) [inline templates/app.ini.tmpl]: I'd prefer using GOGS_<GOGS_PARAM>, so GOGS_ENABLE_REVERSE_PROXY_AUTHENTICATION

File: templates/app.ini.tmpl [hunk 2]
```diff
@@ -18,8 +18,19 @@ REVERSE_PROXY_AUTHENTICATION_USER = X-Forwarded-User
...
+ENABLE_REVERSE_PROXY_AUTHENTICATION     = false
+{{else}}
 ENABLE_REVERSE_PROXY_AUTHENTICATION     = true
+{{end}}
 ENABLE_REVERSE_PROXY_AUTO_REGISTRATION  = true
+{{if not ( eq ( getenv "GOGS_SERVICE_DISABLE_REGISTRATION" ) "" ) }}
```
Comment (mcanevet) [inline templates/app.ini.tmpl]: same here

**Comment by mcanevet** [comment]
We used to use GOGS_section_variable

File: Dockerfile [hunk 3]
```diff
@@ -5,5 +5,7 @@ ADD ./templates /etc/confd/templates
 
 VOLUME ["/app/gogs/custom/conf"]
 
+ENV GOGS_SERVICE_ENABLE_REVERSE_PROXY_AUTHENTICATION=false GOGS_SERVICE_ENABLE_REVERSE_PROXY_AUTO_REGISTRATION=false GOGS_SERVICE_DISABLE_REGISTRATION=false GOGS_SERVICE_SHOW_REGISTRATION_BUTTON=true
```
Comment (mcanevet) [inline Dockerfile]: You can set each environment variable on its line:

```
ENV GOGS_SERVICE_ENABLE_REVERSE_PROXY_AUTHENTICATION false
ENV GOGS_SERVICE_ENABLE_REVERSE_PROXY_AUTO_REGISTRATION false
ENV GOGS_SERVICE_DISABLE_REGISTRATION false
ENV GOGS_SERVICE_SHOW_REGISTRATION_BUTTON true
```

Comment (mcanevet) [inline Dockerfile]: Actually the documentation recommends using your way. Could you please add '\' after each env?

Comment (dabelenda) [inline Dockerfile]: The reason why doing it the way I did is recommended, is because docker creates a layer each time you have a "ENV".

Comment (mcanevet) [inline Dockerfile]: Yep, that what I though and reading the doc confirms that

**Comment by mcanevet** [comment]
LGTM, you can merge, pin and create a new version of the catalog

**Pull request merged**